File: gui/src/types/TitleInfo.ts
export interface TitleInfo {
    ID:           number;
    CreatedAt:    Date;
    UpdatedAt:    Date;
    DeletedAt:    null;
    Title:        string;
    Year:         string;
    Rated:        string;
    Released:     string;
    Runtime:      string;
    Genre:        string;
    Director:     string;
    Writer:       string;
    Actors:       string;
    Plot:         string;
    Language:     string;
    Country:      string;
    Awards:       string;
    Poster:       string;
    imdbRating:   string;
    imdbID:       string;
    Type:         TitleType;
    totalSeasons: string;
    Folder:       string;
}

type TitleType = |"series" | "movie"

File: internal/domain/models/title_info.go
package models

import "gorm.io/gorm"

type TitleInfo struct {
	gorm.Model

	Title        string `json:"Title"`
	Year         string `json:"Year"`
	Rated        string `json:"Rated"`
	Released     string `json:"Released"`
	Runtime      string `json:"Runtime"`
	Genre        string `json:"Genre"`
	Director     string `json:"Director"`
	Plot         string `json:"Plot"`
	Poster       string `json:"Poster"`
	ImdbRating   string `json:"imdbRating"`
	ImdbID       string `json:"imdbID"`
	Type         string `json:"Type"`
	TotalSeasons string `json:"totalSeasons"`
	Folder       string `json:"Folder"`
}
